packages/ihe-gateway/server/Channels/CareQuality Document Exchange - Inbound/XCA ITI-38 Inbound Interface/sourceConnector-transformer-step-1-Payload validation.js

```diff
@@ -1,15 +1,11 @@
+logger.info("sourceConnector-transformer-step-1-Payload validation.js");
 var payload = null, error = null;
 
 if (msg.*::Body.*::AdhocQueryRequest.length() > 0) {
 	payload = msg.*::Body.*::AdhocQueryRequest;
-	destinationSet.removeAllExcept('XCA ITI-38 Inbound Processor');
 	
 } else if (msg.*::Body.*::RetrieveDocumentSetRequest.length() > 0) {
-	payload = msg.*::Body.*::RetrieveDocumentSetRequest;
-	destinationSet.removeAllExcept('XCA ITI-39 Inbound Processor');
-	
-} else {
-	error = 'Unknown request type';
+	error = 'ITI-39 Request on ITI-38 Endpoint';
```

Comment:
We split ITI38 and ITI39 into two different channels because we are using the single channel per response logic so that everything can be synchronous when responding to inbound requests. So we are no longer routing requests from one XCA interface channel to two different XCA processors channels 

The issue I ran up against is that I couldnt turn off ITI38 and ITI39 channels respectively from accepting the hardcoded url path for the other channel. So the only way around this I have is to throw an error if a request comes to the wrong channel.